Bot creates new category on restart

## Expected Behavior
On a restart, I expect it to reuse the existing channels

## Actual Behavior
It creates a new channel

![image](https://user-images.githubusercontent.com/11667059/37377555-877bb42e-2722-11e8-8e92-f2a7f337e203.png)


## Steps to Reproduce the Problem

  1. Start a chat
  1. restart the bot
  1. Talk
  1. ???
  1. Profit

## Specifications

  - Platform: Ubuntu
  - Version (can be found in package.json): 2.4.10
  - Node.js version (`node -v`): 8.10.0

Okay, @Bjornskjald, so I am now currently on d06d6100abe90cc944b4fc2d2b0a095106fdf8a5 and this is what I get:

```
Mar 20 10:32:13 vps systemd[1]: Started Miscord Relay.
Mar 20 10:32:14 vps node[21287]: module.js:549
Mar 20 10:32:14 vps node[21287]:     throw err;
Mar 20 10:32:14 vps node[21287]:     ^
Mar 20 10:32:14 vps node[21287]: Error: Cannot find module 'superagent'
Mar 20 10:32:14 vps node[21287]:     at Function.Module._resolveFilename (module.js:547:15)
Mar 20 10:32:14 vps node[21287]:     at Function.Module._load (module.js:474:25)
Mar 20 10:32:14 vps node[21287]:     at Module.require (module.js:596:17)
Mar 20 10:32:14 vps node[21287]:     at require (internal/module.js:11:18)
Mar 20 10:32:14 vps node[21287]:     at Object.<anonymous> (/opt/miscord/lib/updateNotifier.js:2:20)
Mar 20 10:32:14 vps node[21287]:     at Module._compile (module.js:652:30)
Mar 20 10:32:14 vps node[21287]:     at Object.Module._extensions..js (module.js:663:10)
Mar 20 10:32:14 vps node[21287]:     at Module.load (module.js:565:32)
Mar 20 10:32:14 vps node[21287]:     at tryModuleLoad (module.js:505:12)
Mar 20 10:32:14 vps node[21287]:     at Function.Module._load (module.js:497:3)
Mar 20 10:32:14 vps systemd[1]: miscord.service: Main process exited, code=exited, status=1/FAILURE
Mar 20 10:32:14 vps systemd[1]: miscord.service: Unit entered failed state.
Mar 20 10:32:14 vps systemd[1]: miscord.service: Failed with result 'exit-code'.
```

This is the log created when I run `npm start`

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'start' ]
2 info using npm@5.7.1
3 info using node@v8.10.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle messenger-discord@2.6.6~prestart: messenger-discord@2.6.6
6 info lifecycle messenger-discord@2.6.6~start: messenger-discord@2.6.6
7 verbose lifecycle messenger-discord@2.6.6~start: unsafe-perm in lifecycle true
8 verbose lifecycle messenger-discord@2.6.6~start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/opt/miscord/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
9 verbose lifecycle messenger-discord@2.6.6~start: CWD: /opt/miscord
10 silly lifecycle messenger-discord@2.6.6~start: Args: [ '-c', 'node index.js' ]
11 silly lifecycle messenger-discord@2.6.6~start: Returned: code: 1  signal: null
12 info lifecycle messenger-discord@2.6.6~start: Failed to exec start script
13 verbose stack Error: messenger-discord@2.6.6 start: `node index.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:285:16)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at EventEmitter.emit (events.js:214:7)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at ChildProcess.emit (events.js:214:7)
13 verbose stack     at maybeClose (internal/child_process.js:925:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
14 verbose pkgid messenger-discord@2.6.6
15 verbose cwd /opt/miscord
16 verbose Linux 2.6.32-042stab123.9
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "start"
18 verbose node v8.10.0
19 verbose npm  v5.7.1
20 error code ELIFECYCLE
21 error errno 1
22 error messenger-discord@2.6.6 start: `node index.js`
22 error Exit status 1
23 error Failed at the messenger-discord@2.6.6 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

```jay@vps:/opt/miscord$ sudo npm install
npm ERR! code 1
npm ERR! Command failed: /usr/bin/git clone -q git://github.com/schmavery/facebook-chat-api.git /home/jay/.npm/_cacache/tmp/git-clone-6b798284
npm ERR! /home/jay/.npm/_cacache/tmp/git-clone-6b798284/.git: Permission denied
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jay/.npm/_logs/2018-03-20T12_28_41_809Z-debug.log```

--- a/lib/login/discord.js
+++ b/lib/login/discord.js
@@ -24,7 +24,7 @@ module.exports = config => {
     log.verbose('login', 'Found Discord guild')
 
     // get category and list of channels
-    var category = config.discord.guild.channels.find(channel => (channel.name === config.discord.category || channel.id === config.discord.category) && channel.type === null)
+    var category = config.discord.guild.channels.find(channel => (channel.name === config.discord.category || channel.id === config.discord.category) && (channel.type === null || channel.type === 'category'))
     if (!category) category = await config.discord.guild.createChannel(config.discord.category, 'category')
     log.verbose('login', 'Got Discord category')
     log.silly('login', category)

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "messenger-discord",
-  "version": "2.6.7",
+  "version": "2.6.8",
   "description": "Facebook Messenger to Discord bridge",
   "main": "index.js",
   "scripts": {

